client/my-sites/themes/themes-magic-search-card/index.jsx

```diff
@@ -0,0 +1,110 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import debounce from 'lodash/debounce';
+
+/**
+ * Internal dependencies
+ */
+import Search from 'components/search';
+import SegmentedControl from 'components/segmented-control';
+import { trackClick } from '../helpers';
+import config from 'config';
+import { isMobile } from 'lib/viewport';
+
+const ThemesMagicSearchCard = React.createClass( {
+	propTypes: {
+		tier: React.PropTypes.string,
+		select: React.PropTypes.func.isRequired,
+		site: React.PropTypes.oneOfType( [
+			React.PropTypes.object,
+			React.PropTypes.bool
+		] ).isRequired,
+		onSearch: React.PropTypes.func.isRequired,
+		search: React.PropTypes.string
+	},
+
+	trackClick: trackClick.bind( null, 'search bar' ),
+
+	componentWillMount() {
+		this.onResize = debounce( () => {
+			this.setState( { isMobile: isMobile() } );
+		}, 250 );
+	},
+
+	componentDidMount() {
+		window.addEventListener( 'resize', this.onResize );
+	},
+
+	componentWillUnmount() {
+		window.removeEventListener( 'resize', this.onResize );
+	},
+
+	getInitialState() {
+		return {
+			isMobile: isMobile(),
+			searchIsOpen: false
+		};
+	},
+
+	getDefaultProps() {
+		return { tier: 'all' };
+	},
+
+	onSearchOpen() {
+		this.setState( { searchIsOpen: true } );
+	},
+
+	onSearchClose() {
+		this.setState( { searchIsOpen: false } );
+	},
+
+	onBlur() {
+		if ( this.state.isMobile ) {
+			this.setState( { searchIsOpen: false } );
+		}
+	},
+
+	render() {
+		const isJetpack = this.props.site && this.props.site.jetpack;
```

Comment:
If it will be the common consensus that we should do this way I don't mind and will follow the rest. On personal note I just don't feel that one type specifier for all declarations is good. I like when each declaration has type specified because I don't need to scroll(or at least limit the scroll) to see what type it is, and second is that I don't like the additional indentation required for one type specifier. That being said I know that the biggest value comes from consistency so maybe code guidelines should have more direct regulations here - so I will not be tempted to do it my way :)